kafka-rest/src/test/java/io/confluent/kafkarest/integration/ClusterTestHarness.java

```diff
@@ -565,8 +566,12 @@ protected final void produceJsonMessages(List<ProducerRecord<Object, Object>> re
           final KafkaProducer<T, T> producer = createProducer.get();
           boolean sent = false;
           try {
-            producer.send(rec).get();
-            sent = true;
+            RecordMetadata result = producer.send(rec).get();
+            if (result.hasOffset()) {
+              sent = true;
+            } else {
+              log.info("Failed to get offset back from produce {}", result);
```

Comment:
nit: Maybe also log the originally sent record? If there's no offset maybe it's hard to identify what does this result correspond to.